### PR TITLE
init mnist

### PR DESCRIPTION
**Pull request by jacquesqiao**
fix: https://github.com/PaddlePaddle/Paddle/issues/3557 use low level API to construct a mnist training logic

1. fix `sgd-op` **[done]** https://github.com/PaddlePaddle/Paddle/pull/3556/commits/8b3d33a055b2a1556adedeb41a16b794249a3848
2. add `rowwiseadd-grad-op` **[done]** https://github.com/PaddlePaddle/Paddle/pull/3183 https://github.com/PaddlePaddle/Paddle/pull/3183/commits/82b820e97b90f21d7b46629bba72436a69e888e1
3. add `mul-grad-op` **[done]** https://github.com/PaddlePaddle/Paddle/pull/3379
4. fix `math matmul` **[done]** https://github.com/PaddlePaddle/Paddle/issues/3577
5. fix `rowwiseadd-grad-op` **[done]** https://github.com/PaddlePaddle/Paddle/issues/3604
6. register` rowwise_grad_gpu_op` **[done]** https://github.com/PaddlePaddle/Paddle/pull/3660

### CODE REVIEW

File: python/paddle/v2/framework/tests/mnist.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    cost_name = 'cross_entropy_%d' % uniq_id()
+    cross_entropy_op = Operator(
+        "onehot_cross_entropy", X=input, label=label, Y=cost_name)
+    net.add_op(cross_entropy_op)
+    scope.new_var(cost_name)
+    net.infer_shape(scope)
+    return cost_name
+
+
+def get_backward_net(forward_net):
```
Comment (wangkuiyi) [inline python/paddle/v2/framework/tests/mnist.py]: create_backward_net

File: python/paddle/v2/framework/tests/mnist.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    for input in net.inputs()["all"]:
+        var = scope.new_var(input)
+        var.get_tensor()
+    for output in net.outputs()["all"]:
+        var = scope.new_var(output)
+        var.get_tensor()
+    return net
+
+
+def print_inputs_outputs(op):
```
Comment (wangkuiyi) [inline python/paddle/v2/framework/tests/mnist.py]: debug_print_op

File: python/paddle/v2/framework/tests/mnist.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        learning_rate=learning_rate)
+    net.add_op(optimize_op)
+
+
+# should use operator and add these to the init_network
+def init_param(param_name, dims):
+    var = scope.new_var(param_name)
+    tensor = var.get_tensor()
+    tensor.set_dims(dims)
+    data = numpy.random.uniform(
```
Comment (dzhwinter) [inline python/paddle/v2/framework/tests/mnist.py]: as a demo, maybe we better use the built-in operator?

Comment (jacquesqiao) [inline python/paddle/v2/framework/tests/mnist.py]: done

File: python/paddle/v2/framework/tests/mnist.py [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+def set_cost(cost):
+    cost_shape = numpy.array(scope.find_var(cost).get_tensor()).shape
+    cost_grad = \
+        scope.find_var(grad_var_name(cost)).get_tensor()
+    cost_grad.set_dims(cost_shape)
+    cost_grad.alloc_float(place)
+    cost_grad.set(numpy.ones(cost_shape).astype("float32"), place)
+
+
+def mean_cost(cost):
```
Comment (dzhwinter) [inline python/paddle/v2/framework/tests/mnist.py]: mean cost get the cross_entropy output for debugging. we'd better rename this snippet.

Comment (jacquesqiao) [inline python/paddle/v2/framework/tests/mnist.py]: done

**Pull request merged**